### PR TITLE
Update E2E tests to work with WP 6.7 RC2

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Bump WordPress version used for E2E tests to 6.7 RC2 (go back to `wordpress-latest` once 6.7 is released) and adjust the tests

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. CI should be green.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection.block_theme.spec.ts [hunk 1]
```diff
@@ -73,15 +74,11 @@ test.describe( 'Product Collection: Register Product Collection', () => {
...
-		const productCollectionBlock = editor.canvas.getByLabel(
-			'Block: Product Collection'
-		);
-
 		for ( const myCollection of Object.values(
 			MY_REGISTERED_COLLECTIONS
 		) ) {
 			await expect(
-				productCollectionBlock.getByRole( 'button', {
+				page.getByRole( 'button', {
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/tests/e2e/tests/product-collection/register-product-collection.block_theme.spec.ts:81]: Collections are displayed in `components-popover`. It used to be available from canvas but apparently the way it is rendered or place it is anchored changed and should be looked for on `page`.

File: plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts [hunk 2]
```diff
@@ -73,7 +73,12 @@ export class Editor extends CoreEditor {
 
 	async revertTemplate( { templateName }: { templateName: string } ) {
 		await this.page.getByPlaceholder( 'Search' ).fill( templateName );
-		await this.page.getByLabel( templateName, { exact: true } ).click();
+		// WP 6.7 issue: https://github.com/WordPress/gutenberg/issues/66585.
+		// Template-parts have aria-label="[object Object]" hence switching
+		// from using label to button.
+		await this.page
+			.getByRole( 'button', { name: templateName, exact: true } )
+			.click();
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/tests/e2e/utils/editor/editor-utils.page.ts:81]: This is essentially a switch from using preview (red) to title (green) to access the template
Screen Shot 2024-10-30 at 11 54 14 AM

**Review by mdperez86 — APPROVED** [review]
This is working as expected.
Nice job @kmanijak !!!

**Review by gigitux — APPROVED** [review]
LGTM! Thanks for working on it!

I'm going to merge this PR, given that will fix some CI errors!

**Comment by adimoldovan** [comment]
@gigitux @kmanijak just wanted to double check switching to wordpress-6.7-RC2.zip as default.
Will these fixes work with both latest and 6.7, or will these tests fail for latest now?
Blocks e2e tests are running daily with the RC anyway, but with this merge the latest version is not covered anymore.

**Comment by gigitux** [comment]
> @gigitux @kmanijak just wanted to double check switching to wordpress-6.7-RC2.zip as default. Will these fixes work with both latest and 6.7, or will these tests fail for latest now? Blocks e2e tests are running daily with the RC anyway, but with this merge the latest version is not covered anymore.

We currently run our E2E test suite only on a specific version of WordPress. During the last two or three releases, we’ve switched to the next version two or three weeks before its release. This approach ensures that contributors can continue to merge PRs seamlessly after a new WordPress release.

**Comment by kmanijak** [comment]
> We currently run our E2E test suite only on a specific version of WordPress. During the last two or three releases, we’ve switched to the next version two or three weeks before its release. This approach ensures that contributors can continue to merge PRs seamlessly after a new WordPress release.

On top of @gigitux's comment, once WP 6.7 is released we should switch the version back to `wordpress-latest`.

**Pull request merged**